latex/epoch.tex

```diff
@@ -1,54 +1,614 @@
-The pool reaping in rules are discribed in
-Figure~\ref{fig:ts-types:pool-reap} and Figure~\ref{fig:rules:pool-reap}.
+\newcommand{\UTxOEpEnv}{\type{UTxOEpEnv}}
+\newcommand{\UTxOEpState}{\type{UTxOEpState}}
+\newcommand{\Accnt}{\type{Accnt}}
+\newcommand{\AccntEnv}{\type{AccntEnv}}
+\newcommand{\AccntState}{\type{AccntState}}
+\newcommand{\StPlCleanEnv}{\type{StPlCleanEnv}}
+\newcommand{\StPlCleanState}{\type{StPlCleanState}}
+\newcommand{\NewProtoConstsEnv}{\type{NewProtoConstsEnv}}
+\newcommand{\NewProtoConstsState}{\type{NewProtoConstsState}}
+\newcommand{\EpochEnv}{\type{EpochEnv}}
+\newcommand{\EpochState}{\type{EpochState}}
+\newcommand{\Production}{\type{Production}}
+
+\newcommand{\obligation}[4]{\fun{obligation}~ \var{#1}~ \var{#2}~ \var{#3}~ \var{#4}}
+\newcommand{\reward}[5]{\fun{reward}~ \var{#1}~ \var{#2}~ \var{#3}~ \var{#4}~ \var{#5}}
+\newcommand{\poolRefunds}[3]{\fun{poolRefunds}~ \var{#1}~ \var{#2}~ \var{#3}}
 
 %%
-%% Figure - Pool Reap
+%% Figure - Epoch Abstract Types
 %%
+\begin{figure}
+  \emph{Abstract types}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \var{production} & \Production & \text{blocks produced in the epoch}\\
+    \end{array}
+  \end{equation*}
+  \caption{Epoch definitions}
+  \label{fig:epoch-defs}
+\end{figure}
+
+%%
+%% Figure - Functions for Epoch Rules
+%%
+\begin{figure}
+  \begin{align*}
+      & \fun{obligation} \in \PrtclConsts \to \Allocs \to \Allocs \to \Slot \to \Coin
+      & \text{total possible refunds} \\
+      & \obligation{pc}{stkeys}{stpools}{cslot} =\\
+      & \sum\limits_{(\_ \mapsto s) \in \var{stkeys}}
+        \refund{k_{\mathsf{val}}}{k_{\min}}{(\slotminus{cslot}{s})}{\lambda} \\
+      & + \sum\limits_{(\_ \mapsto s) \in \var{stpools}}
+        \refund{p_{\mathsf{val}}}{p_{\min}}{(\slotminus{cslot}{s})}{\lambda} \\
+      &
+        \where k_{\mathsf{val}}, k_{\min},p_{\mathsf{val}}, p_{\min}, \lambda \in pc
+      \nextdef
+      & \fun{reward} \in \Production \to \PrtclConsts \to \Coin \to DWState \to \\
```

Comment:
If we have a function that depends on a list of blocks, then we're loosing the spirit of the small-step style. This is the remark I got from @dcoutts when I defined a function on a list of transactions. Could the rewards be calculated more "operationally"?

BTW, this could be a problem, as Neil pointed out, if we cannot put a bound to the computation of rewards.

Reply:
this is a really good point, this should be broken down into steps.  I'll wait to do this, though, until the details are back from the researchers as to what this reward function will be.  (It is described in the delegation doc, but it may have to be changed based on some tbd details about new versions of ouroboros.)